ghost/email-service/lib/EmailRenderer.js

```diff
@@ -715,13 +724,18 @@ class EmailRenderer {
      * @param {object} postModel
      * @returns
      */
-    #getEmailPreheader(postModel) {
+    #getEmailPreheader(postModel, addPaywall, html) {
         let plaintext = postModel.get('plaintext');
         let customExcerpt = postModel.get('custom_excerpt');
         if (customExcerpt) {
             return customExcerpt;
         } else {
             if (plaintext) {
+                // If paywall is enabled, we need to get the plaintext from the html
+                // as the plaintext field on the model may contain paywalled content
+                if (addPaywall) {
+                    plaintext = htmlToPlaintext.email(html);
+                }
```

Comment:
I think we need to check for segment == `status:free` instead of whether there is a paywall or not, because it is also possible to have content that is only visible for paid members when there is no paywall card in the post.

Reply:
👍 Updated in https://github.com/TryGhost/Ghost/pull/17137/commits/553fadb1f31277b5f5f92d1318cb9c6663707c4b